apps/captiveportal/qml/pages/CaptivePortalPage.qml

```diff
@@ -116,7 +116,7 @@ Page {
         fullscreenHeight: portrait ? Screen.height : Screen.width
         portrait: browserPage.isPortrait
         maxLiveTabCount: 3
-        toolbarHeight: overlay.toolBar.height
+        toolbarHeight: Math.min(overlay.toolBar.height, webView.fullscreenHeight - overlay.yPosition)
```

Comment:
Similarly as commented here https://github.com/sailfishos/sailfish-browser/pull/741/files#r640564376 we should not animate these things. Further, toolbarHeight is a fixed value for a reason.

Overall this PR looks too complex for the thing that it is trying achieve.